lib/whois/record/parser/whois.tld.ee.rb

```diff
@@ -20,9 +18,111 @@ class Parser
       # @see Whois::Record::Parser::Example
       #   The Example parser for the list of all available methods.
       #
-      class WhoisTldEe < BaseWhoisd
-      end
+      class WhoisTldEe < Base
+        include Scanners::Scannable
+        class_attribute :status_mapping
+
+        self.status_mapping = {
+          'ok (paid and in zone)' => :registered,
+          'expired' => :expired
+        }
+
+        self.scanner = Scanners::WhoisTldEe
+
+        property_supported :available? do
+          !!node('status:available')
+        end
+
+        property_supported :registered? do
+          !available?
+        end
+
+        property_supported :domain do
+          if (content_for_scanner =~ /^Domain:\nname:\s+(.+)\n/)
+            $1.to_s.strip.downcase
+          end
+        end
+
+        property_not_supported :domain_id
+
+        property_supported :status do
+          if content_for_scanner =~ /status:\s+(.+?)\n/
+            self.class.status_mapping[$1] || $1
+          else
+            :available
+          end
+        end
+
+        property_supported :created_on do
+          if content_for_scanner =~ /registered:\s+(.+?)\n/
+            Time.parse($1)
+          end
+        end
+
+        property_supported :updated_on do
+          if content_for_scanner =~ /changed:\s+(.+?)\n/
+            Time.parse($1)
+          end
+        end
 
+        property_supported :expires_on do
+          if content_for_scanner =~ /expire:\s+(.+?)\n/
+            Time.parse($1)
+          end
+        end
+
+        property_supported :disclaimer do
+          node('field:disclaimer')
+        end
+
+        property_supported :registrant_contacts do
+          build_contact('Registrant', Whois::Record::Contact::TYPE_REGISTRANT)
+        end
+
+        property_supported :admin_contacts do
+          build_contact('Administrative contact', Whois::Record::Contact::TYPE_ADMINISTRATIVE)
+        end
+
+        property_supported :technical_contacts do
+          build_contact('Technical contact', Whois::Record::Contact::TYPE_TECHNICAL)
+        end
+
+        property_supported :registrar do
+          node('Registrar') do |hash|
+            Record::Registrar.new(
+              :name         => hash['name'],
+              :organization => hash['name'],
+              :url          => hash['url']
+            )
+          end
+        end
+
+        property_supported :nameservers do
+          node('Name servers') do |hash|
+            Array.wrap(hash['nserver']).map do |name|
+              Nameserver.new(:name => name.downcase)
```

Comment:
I'd prefer to use the new Hash syntax here:

    Nameserver.new(name: name.downcase)